src/ICSharpCode.SharpZipLib/Zip/ZipInputStream.cs

```diff
@@ -612,7 +687,7 @@ private int BodyRead(byte[] buffer, int offset, int count)
 
 			bool finished = false;
 
-			switch (method)
+			switch (entry.CompressionMethod)
```

Comment:
Codacy found an issue: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/piksel/SharpZipLib/pullRequest?prid=5745488)